src/vs/workbench/contrib/chat/browser/chatViewPane.ts

```diff
@@ -317,4 +351,56 @@ export class ChatViewPane extends ViewPane implements IViewWelcomeDelegate {
 			(this.viewState as Record<string, unknown>)[key] = value;
 		}
 	}
+
+	private updateViewTitle(model?: IChatModel): void {
+		model = model ?? this._widget.viewModel?.model;
+		const title = (model && model.hasCustomTitle) ? model.title : localize('chat', "Chat");
+
+		this.updateTitle(localize('chat', "Chat"));
+		if (model && model.hasCustomTitle) {
+			this.setSecondaryTitle(title);
+		} else if (model && model.getRequests().length > 0) {
+			this.setSecondaryTitle(localize('generatingTitle', "Generating title..."));
+		} else {
+			this.setSecondaryTitle(undefined);
+		}
+	}
+
+	private setSecondaryTitle(title: string | undefined): void {
+		if (!this._secondaryTitleContainer || !this._secondaryTitle) {
+			return;
+		}
+		if (title) {
+			this._secondaryTitle.textContent = title;
+			this._secondaryTitleContainer.style.display = 'flex';
+		} else {
+			this._secondaryTitleContainer.style.display = 'none';
+		}
+		if (this._lastLayout) {
+			this.layoutBody(this._lastLayout.height, this._lastLayout.width);
+		}
+	}
 }
+
+registerAction2(class BackToChatAction extends Action2 {
```

Comment:
This can go with other actions in one of the chatActions.ts files

Reply:
As mentioned elsewhere, I removed the back to chat action, but will keep this in mind if/when we add it back :)